pkg/build/util/generate.go

```diff
@@ -30,3 +32,43 @@ func GenerateBuildFromBuild(build *api.Build) *api.Build {
 		},
 	}
 }
+
+// SubstituteImageReferences replaces references to an image with a new value
+func SubstituteImageReferences(build *api.Build, oldImage string, newImage string) {
+	switch {
+	case build.Parameters.Strategy.Type == api.DockerBuildStrategyType &&
+		build.Parameters.Strategy.DockerStrategy != nil &&
+		build.Parameters.Strategy.DockerStrategy.BaseImage == oldImage:
+		build.Parameters.Strategy.DockerStrategy.BaseImage = newImage
+	case build.Parameters.Strategy.Type == api.STIBuildStrategyType &&
+		build.Parameters.Strategy.STIStrategy != nil &&
+		build.Parameters.Strategy.STIStrategy.Image == oldImage:
+		build.Parameters.Strategy.STIStrategy.Image = newImage
+	case build.Parameters.Strategy.Type == api.CustomBuildStrategyType:
+		// update env variable references to the old image with the new image
+		if build.Parameters.Strategy.CustomStrategy.Env == nil {
```

Comment:
please do `strategy := build.Parameters.Strategy.CustomStrategy` at the beginning of this block.... it would make this block readable by humans and potentially very smart dogs.